.github/workflows/rust.yml

```diff
@@ -395,8 +364,8 @@ jobs:
           key: ${{ runner.os }}-cargo-miri-${{ hashFiles('**/Cargo.lock') }}
       - name: Setup Rust toolchain
         run: |
-          rustup toolchain install ${{ matrix.rust }}
-          rustup default ${{ matrix.rust }}
+          rustup toolchain install nightly-2022-01-17
```

Comment:
why use this version？
Is there a possibility that this version needs to be updated later?

Reply:
I think we probably pinned the version a while ago when some nightly had issues. I think updating it would be a great idea (and as you say a good follow on PR)

Reply:
Follow on PR: https://github.com/apache/arrow-datafusion/pull/3215 (will need some work)